ogr/ogrsf_frmts/miramon/ogrmiramon.h

```diff
@@ -0,0 +1,173 @@
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE. 
+ ****************************************************************************/
+
+#ifndef OGRMIRAMON_H_INCLUDED
+#define OGRMIRAMON_H_INCLUDED
+
+#include "ogrsf_frmts.h"
+#include "ogr_api.h"
+#include "cpl_string.h"
+#include "mm_wrlayr.h"
+
+/************************************************************************/
+/*                             OGRMiraMonLayer                          */
+/************************************************************************/
+
+class OGRMiraMonLayer final
+    : public OGRLayer,
+      public OGRGetNextFeatureThroughRaw<OGRMiraMonLayer>
+{
+    OGRSpatialReference *m_poSRS = nullptr;
+    OGRFeatureDefn *poFeatureDefn;
+
+    GUIntBig iNextFID;
+
+    // Pointer to one of three possible MiraMon layers: points,
+    // arcs or polygons. Every time a feature is read this pointer
+    // points to the appropiate layer
+    struct MiraMonVectLayerInfo *phMiraMonLayer;
+
+    // When writing a layer
+    struct MiraMonVectLayerInfo hMiraMonLayerPNT;  // MiraMon points layer
+    struct MiraMonVectLayerInfo hMiraMonLayerARC;  // MiraMon arcs layer
+    struct MiraMonVectLayerInfo hMiraMonLayerPOL;  // MiraMon polygons layer
+
+    // When reading a layer or the result of writing is only a DBF
+    struct MiraMonVectLayerInfo hMiraMonLayerReadOrNonGeom;
+
+    struct MiraMonFeature hMMFeature;  // Feature reading/writing
+
+    bool bUpdate;
+
+    // Ratio used to enhance certain aspects of memory
+    // In some memory settings, a block of 256 or 512 bytes is used.
+    // This parameter can be adjusted to achieve
+    // nMemoryRatio*256 or nMemoryRatio*512.
+    // For example, nMemoryRatio=2 in powerful computers and
+    // nMemoryRatio=0.5 in less powerful computers.
+    // By increasing this parameter, more memory will be required,
+    // but there will be fewer read/write operations to the disk.
+    double nMMMemoryRatio;
+
+    VSILFILE *m_fp = nullptr;
+
+    // Array of string or doubles used in the field features processing
+    char **papszValues;
+    double *padfValues;
```

Comment:
as we are here in C++ land, you could use a std::vector for all those arrays, and enjoy automatic memory management, but I don't make this a required change (at least at that stage of my code review :-)). More in the "for your information" category.

Reply:
Ok, thanks. Next time I'll do something with arrays in C++ land I'll use them!

Reply:
> Ok, thanks. Next time I'll do something with arrays in C++ land I'll use them!

actually I had to remove the char **papszValues use in b66006e8c21d27a90198f0dc2c4036869bb43d67 and replace it by a local use of CPLStringList, as this was causing memory leaks.